internal/service/advancedclustertpf/resource_compatibility_reuse.go

```diff
@@ -3,14 +3,49 @@ package advancedclustertpf
 import (
 	"context"
 	"fmt"
+	"strconv"
 	"strings"
 
 	"github.com/mongodb/terraform-provider-mongodbatlas/internal/common/constant"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/common/conversion"
 	"github.com/spf13/cast"
 	admin20240530 "go.mongodb.org/atlas-sdk/v20240530005/admin"
 	"go.mongodb.org/atlas-sdk/v20241113002/admin"
 )
 
+type MajorVersionOperator int
+
+const (
+	EqualOrHigher MajorVersionOperator = iota
+	Higher
+	EqualOrLower
+	Lower
+)
+
+func MajorVersionCompatible(input *string, version float64, operator MajorVersionOperator) *bool {
+	if !conversion.IsStringPresent(input) {
+		return nil
+	}
+	value, err := strconv.ParseFloat(*input, 64)
+	if err != nil {
+		return nil
+	}
+	var result bool
+	switch operator {
+	case EqualOrHigher:
+		result = value >= version
+	case Higher:
+		result = value > version
```

Comment:
nit: The < and > operators are called "less than" and "greater than" operators, respectively. in case we want to use Less and Greater instead of Lower and Higher